pkg/dbmate/db.go

```diff
@@ -85,28 +85,44 @@ func (db *DB) Wait() error {
 	return fmt.Errorf("unable to connect to database: %s", err)
 }
 
-// CreateAndMigrate creates the database (if necessary) and runs migrations
-func (db *DB) CreateAndMigrate() error {
+// CreateAndMigrate creates the database (if necessary) and runs migrations.
+// If withRollback is true, this doesn't create or migrate but
+// returns if creation of database and applying of migrations would succeed.
+func (db *DB) CreateAndMigrate(withRollback bool) error {
```

Comment:
Just create a new function for this. Someone mis-placing the arg here could drop their DB. Probably not what they want. The arg is misleading because dropping a DB cannot be rolled back on any RDBMS I know of.

The goal seems to be to experiment with running a migration. A safer way to do that is to dump the CREATE statement for the current schema (typically requires querying metadata tables), and then creating a temporary table with a synthesized name and no rows...run the migration on *that* if you want to drop it. You probably never want to hide a DROP inside a function if a user actually has data in a table etc.

Reply:
Good points. I will split this into a separate function. 

Btw, I want to clarify that this function doesn't support `MigrateAndDrop`. It only drops if it creates the database here _and_ `withRollback` was true.

Reply:
If you feel that the design of even supporting withRollback for Creation part of Create and migrate is dangerous, I agree and I'm willing to remove that functionality. For my scenario, I only care about "withRollback" for migrations and not at all about creation.

Reply:
done.